### DISCUSSION TITLE
멀티플레이 캐릭터에 직접 만든 컴포넌트를 붙이는법

### DISCUSSION THREAD

**Discussion by JeongYire** [naverz/zepeto-studio-kor#261] [5. Q&A]
현재 멀티플레이어 예제 스크립트를 사용하고 잇는데
OnTriggerEnter 를 통해서 캐릭터의 gameobject 는 갖고 올 수 있는데

그 gameObject로 그 캐릭터가 어떤 사람이 조종하는지 알고 싶습니다.
그래서 아예 컴포넌트를 추가해서 거기다가 조종하는 유저의 정보( 세션아이디 같은것들 )를 넣어두고 
충돌할때마다 조회하고싶은데 

어디부분에 추가해야 멀티플레이 환경에서도 조회가 되나요?

OnAddPlayer 부분에 추가해서 

            if (isFirst) {

            // [CharacterController] (Local)Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
            ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
                const myPlayer = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer;
                myPlayer.character.gameObject.AddComponent<TextPlayer>().player = myPlayer;
                myPlayer.character.OnChangedState.AddListener((cur, next) => {
                    this.SendState(next);
                });
            });

            // [CharacterController] Player 인스턴스가 Scene에 완전히 로드되었을 때 호출
            ZepetoPlayers.instance.OnAddedPlayer.AddListener((sessionId: string) => {
                const isLocal = this.room.SessionId === sessionId;
                
                if (!isLocal) {
                    const player: Player = this.currentPlayers.get(sessionId);

                    ZepetoPlayers.instance.GetPlayer(sessionId).character.gameObject.AddComponent<TextPlayer>().player = ZepetoPlayers.instance.GetPlayer(sessionId);
                    
                    // [RoomState] player 인스턴스의 state가 갱신될 때마다 호출됩니다.
                    player.OnChange += (changeValues) => this.OnUpdatePlayer(sessionId, player);
                }else{
                    ZepetoPlayers.instance.GetPlayer(sessionId).character.gameObject.AddComponent<TextPlayer>().player = ZepetoPlayers.instance.GetPlayer(sessionId);
                }
            });
        }

이렇게까지 해놨는데 추가가 안됩니다..
혹시 테스트환경 ( 제페토스튜디오 홈페이지에 빌드한후 주소를 뿌려 어플로 돌리고 있을경우 ) 에서는 안되는건가요?

**Comment by x4cf** [comment]
제페토 캐릭터 오브젝트에는 컴포넌트 추가가 잘 안되더라구요. 저는 새로운 오브젝트를 만들고 제페토 캐릭터의 자식으로 달아서 사용합니다 :D

**Reply by JeongYire** [reply]
해결됐습니다 말씀하신대로 addComponent 로 안하고 그냥 게임오브젝트를 자식으로 넣으니 바로 되네요
